src/contexts/walletconnect.tsx

```diff
@@ -115,6 +115,32 @@ export const WalletConnectContextProvider: FC = ({ children }) => {
   const [sessionTopic, setSessionTopic] = useState(initialContext.sessionTopic)
   const [connectedDAppMetadata, setConnectedDappMetadata] = useState(initialContext.connectedDAppMetadata)
 
+  const resetSession = useCallback(() => {
+    setRequiredChainInfo(undefined)
+    setProposalEvent(undefined)
+    setWcSessionState('uninitialized')
+    setSessionTopic(undefined)
+  }, [])
+
+  const tryRestoreLastSession = useCallback(
+    async (client: SignClient) => {
+      if (!client.session.length) return resetSession()
```

Comment:
Note to myself: When checking for active sessions, we should probably first cross-reference them with the active pairings. See: https://github.com/alephium/mobile-wallet/blob/master/src/utils/walletConnect.ts#L50-L58